components/script/dom/customelementregistry.rs

```diff
@@ -456,6 +456,196 @@ impl CustomElementDefinition {
+                    (definition.callbacks.adopted_callback.clone(), args)
+                },
+                CallbackReaction::AttributeChanged(local_name, old_val, val, namespace) => {
+                    // Step 4
+                    if !definition.observed_attributes.iter().any(|attr| *attr == *local_name) {
+                        return;
+                    }
+
+                    let cx = element.global().get_cx();
+
+                    let local_name = DOMString::from(&*local_name);
+                    rooted!(in(cx) let mut name_value = UndefinedValue());
+                    unsafe { local_name.to_jsval(cx, name_value.handle_mut()); }
+
+                    rooted!(in(cx) let mut old_value = NullValue());
+                    if let Some(old_val) = old_val {
+                        unsafe { old_val.to_jsval(cx, old_value.handle_mut()); }
+                    }
+
+                    rooted!(in(cx) let mut value = NullValue());
+                    if let Some(val) = val {
+                        unsafe { val.to_jsval(cx, value.handle_mut()); }
+                    }
+
+                    let namespace = DOMString::from(&*namespace);
+                    rooted!(in(cx) let mut namespace_value = UndefinedValue());
+                    unsafe { namespace.to_jsval(cx, namespace_value.handle_mut()); }
+
+                    let mut args = Vec::with_capacity(4);
+                    args.push(Heap::default());
+                    args.push(Heap::default());
+                    args.push(Heap::default());
+                    args.push(Heap::default());
+                    args.get_mut(0).unwrap().set(name_value.get());
+                    args.get_mut(1).unwrap().set(old_value.get());
+                    args.get_mut(2).unwrap().set(value.get());
+                    args.get_mut(3).unwrap().set(namespace_value.get());
+
+                    (definition.callbacks.attribute_changed_callback.clone(), args)
+                },
+            };
+
+            // Step 3
+            let callback = match callback {
+                Some(callback) => callback,
+                None => return,
+            };
+
+            // Step 5
+            element.push_reaction(CustomElementReaction::Callback(callback, args.into_boxed_slice()));
+
+            // Step 6
+            self.enqueue_element(element);
+        }
+    }
+}
+
+/// https://html.spec.whatwg.org/multipage/#element-queue
+#[derive(HeapSizeOf, JSTraceable)]
+#[must_root]
+struct ElementQueue {
+    queue: DOMRefCell<Vec<JS<Element>>>,
+}
+
+impl ElementQueue {
+    fn new() -> ElementQueue {
+        ElementQueue {
+            queue: Default::default(),
+        }
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#invoke-custom-element-reactions
+    fn invoke_reactions(&self) {
+        // Steps 1-2
+        for element in self.queue.borrow().iter() {
+            element.invoke_reactions()
```

Comment:
I do not believe so, there are two queues an element can be added to:
1. The current element queue.
2. The backup element queue.

For the first case, the current element queue is popped from the custom element reaction stack thus it is no longer the current element queue which means no element will be added to it.

The second case is the backup element queue has the `processing_backup_element_queue` flag which prevents elements being enqueued until the flag is set to `NotProcessing`.